### PR TITLE
Offered template

### LINKED ISSUE #2469: Offer templates in sidebar of new wiki page
We have great templates available for wiki pages: https://publiclab.org/wiki/wiki-templates

They can be used using URLs like: `http://publiclab.org/wiki/new?n=1234` where 1234 is the template node id.

Node id at which particular template exist can be found at https://publiclab.org/wiki/wiki-templates by clicking on the button which creates those template. 

We should offer the main ones on the new wiki page form -- for example, if you're logged in:

https://publiclab.org/wiki/nonexistent-page

image

the template for that is here: 

https://github.com/publiclab/plots2/blob/75db6c9d9b1c02b9ea7e30191ee886b672f966db/app/views/wiki/edit.html.erb#L3-L12

The excerpted section (showing "related") is already only shown when it's being displayed for new wiki pages. So it's a perfect place to put a few links. Maybe something like:

****

### Templates

Use these to start a page from [one of these templates](https://publiclab.org/wiki/wiki-templates):

* [Method template lite]()
* [Method template]()

****

Clicking the link would go to the same URL as you're currently on, but with `?n=_____` appended!

### PR DESCRIPTION
**Pull request by Ronaq13**
Added a link in sidenav to access to template while creating a
new wiki.

close #2469

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts
* [ ] PR is descriptively titled
* [ ] PR body includes `fixes #0000`-style reference to original issue #
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

### CODE REVIEW

**Comment by jywarren** [comment]
Cool!! Can you post a screenshot so we can get a look at it? Thanks!!!

**Comment by Ronaq13** [comment]
@jywarren is it good to be merged now?

File: app/views/wiki/edit.html.erb [hunk 1]
```diff
@@ -21,6 +21,13 @@
       <li><a href="/wiki/authoring-help#Advanced+formatting"><%= t('wiki.edit.advanced_formatting') %></a></li>
     </ul>
   </div>
+  <hr />
+  <div class="sidebar-nav hidden-sm">
+    <ul class="nav nav-list">
+      <li class="nav-header"><h4>Templates</h4></li>
+      <li><a href="/post">Method Template</a></li>
```
Comment (Gauravano) [inline app/views/wiki/edit.html.erb]: >Clicking the link would go to the same URL as you're currently on, but with ?n=_____ appended!

 URL needs to be updated.

File: app/views/wiki/edit.html.erb [hunk 2]
```diff
@@ -21,6 +21,13 @@
       <li><a href="/wiki/authoring-help#Advanced+formatting"><%= t('wiki.edit.advanced_formatting') %></a></li>
     </ul>
   </div>
+  <hr />
+  <div class="sidebar-nav hidden-sm">
+    <ul class="nav nav-list">
+      <li class="nav-header"><h4>Templates</h4></li>
```
Comment (Gauravano) [inline app/views/wiki/edit.html.erb]: ****

### Templates

Use these to start a page from [one of these templates](https://publiclab.org/wiki/wiki-templates):

* [Method template lite]()
* [Method template]()

****
As commented by @jywarren here https://github.com/publiclab/plots2/issues/2469#issue-303990987 , I guess template still needs to be updated to resemble above view. Thanks.

File: app/views/wiki/edit.html.erb [hunk 3]
```diff
@@ -21,6 +21,13 @@
       <li><a href="/wiki/authoring-help#Advanced+formatting"><%= t('wiki.edit.advanced_formatting') %></a></li>
     </ul>
   </div>
+  <hr />
+  <div class="sidebar-nav hidden-sm">
```
Comment (Gauravano) [inline app/views/wiki/edit.html.erb]: Nice work with this, just a few changes and it's ready for merge.Thanks @Ronaq13 .

File: app/views/wiki/edit.html.erb (hunk 1 above)
Comment (Ronaq13) [reply app/views/wiki/edit.html.erb]: I got that. But I have a doubt. We have the same template for 'Method Template' for all the node from 1 to 43. Just title and body of the node get changed on view different node by giving different values to param[:n]. You can also see this in `editor_controller#rich` method.
So I placed the URL for Method Template to `/post?n=1` and for lite it is `/wiki/new`. 
Please tell me if I am wrong or need to update something. Thanks :smile:

**Comment by Ronaq13** [comment]
@Gauravano is it good now?

File: app/views/wiki/edit.html.erb (hunk 1 above)
Comment (Gauravano) [reply app/views/wiki/edit.html.erb]: I guess, there are some updates need to be done on the issue itself. In the issue, it's specified that we want to change the template of the wiki but `/post` is used for the creation of note so I am updating the issue.Also, @Ronaq13 you need to specify `n = 14552` as in production database node ID at which method template lite is there is at 14552. See here https://publiclab.org/wiki/wiki-templates .Also, it's `method template lite ` for which you need to append n=14552 and I  guess method template is the normal one which we use, so, for now, give it URL `/wiki/new`. @jywarren please check if I am right on details. Thanks.

**Comment by Gauravano** [comment]
Left some comments above.Thanks.

File: app/views/wiki/edit.html.erb (hunk 1 above)
Comment (Ronaq13) [reply app/views/wiki/edit.html.erb]: Done

File: app/views/wiki/edit.html.erb [hunk 4]
```diff
@@ -21,6 +21,15 @@
...
+    <p class="nav-header"><h4><b>Templates</b></h4></p>
+    <p>Use these to start a page from one of these templates:</p>
+    <ul>
+      <li><a href="/post?n=14552">Method Template lite</a></li>
```
Comment (Gauravano) [inline app/views/wiki/edit.html.erb]: @Ronaq13 as I mentioned here issue requires the change of wiki template and at `/post` we create new note so please change href to "/wiki/new?n=14552" .Thanks.

Comment (Ronaq13) [reply app/views/wiki/edit.html.erb]: Changed accordingly

Comment (Gauravano) [reply app/views/wiki/edit.html.erb]: Line 29 need to be changed as "/post" is on that, line 30 was fine earlier

**Comment by jywarren** [comment]
Looks like we're almost there -- thanks to you both! 👍 Please ping me when you're done and I'll merge it in. Thanks a lot!

**Comment by jywarren** [comment]
Hi, sorry, just checking -- is this good to go now? Thanks so much!!!

File: app/views/wiki/edit.html.erb [hunk 5]
```diff
@@ -21,6 +21,15 @@
       <li><a href="/wiki/authoring-help#Advanced+formatting"><%= t('wiki.edit.advanced_formatting') %></a></li>
     </ul>
   </div>
+  <hr />
+  <div class="sidebar-nav hidden-sm">
+    <p class="nav-header"><h4><b>Templates</b></h4></p>
+    <p>Use these to start a page from one of these templates:</p>
+    <ul>
+      <li><a href="/post">Method Template lite</a></li>
```
Comment (Gauravano) [inline app/views/wiki/edit.html.erb]: At line 29, link would be `<a href="/wiki/new">Method Template</a>` . hi @jywarren, please check if the link is correct. Thanks.

Comment (jywarren) [reply app/views/wiki/edit.html.erb]: It'll be: `/wiki/new?n=14552&title=Your%20method%20name%20here`

**Comment by SidharthBansal** [comment]
As the person is inactive for more than a month, I am closing the PR. In case you want to push changes please feel free to open a new PR.
Thanks for contributing at Public Lab

**Pull request closed without merging**